src/data/pfun.lean

```diff
@@ -448,4 +451,27 @@ ext $ λ _ _, by simp only [comp_apply, part.bind_comp]
 lemma coe_comp (g : β → γ) (f : α → β) : ((g ∘ f : α → γ) : α →. γ) = (g : β →. γ).comp f :=
 ext $ λ _ _, by simp only [coe_val, comp_apply, part.bind_some]
 
+/-- Product of partial functions. -/
+protected def prod (f : α →. γ) (g : β →. δ) : α × β →. γ × δ :=
+λ x, ⟨(f x.1).dom ∧ (g x.2).dom, λ h, ((f x.1).get h.1, (g x.2).get h.2)⟩
+
+@[simp] lemma dom_prod (f : α →. γ) (g : β →. δ) :
+  (f.prod g).dom = {x | (f x.1).dom ∧ (g x.2).dom} := rfl
+
+lemma get_prod (f : α →. γ) (g : β →. δ) (x : α × β) (h) :
+  (f.prod g x).get h = ((f x.1).get h.1, (g x.2).get h.2) := rfl
+
+@[simp] lemma prod_apply (f : α →. γ) (g : β →. δ) (x : α × β) :
+  f.prod g x = ⟨(f x.1).dom ∧ (g x.2).dom, λ h, ((f x.1).get h.1, (g x.2).get h.2)⟩ := rfl
```

Comment:
I'm not sure that `dom_prod` and `prod_apply` should be `simp` lemmas. In each case RHS looks larger than LHS.

Reply:
Compare to `comp_apply`, `to_subtype_apply`... This is also the only way I got automation to work to prove `prod_id_id` and `prod_comp_comp`.

Reply:
Let me assign this PR to @digama0, since I (almost) never used `part` and `pfun`.